saleor/graphql/account/utils.py

```diff
@@ -173,3 +173,133 @@ def get_groups_which_user_can_manage(user: "User") -> List[Optional[Group]]:
             editable_groups.append(group)
 
     return editable_groups
+
+
+def get_not_manageable_permissions_after_group_deleting(group):
+    """Return permissions that cannot be managed after deleting the group.
+
+    After removing group, for each permission, there should be at least one staff member
+    who can manage it (has both “manage staff” and this permission).
+    """
+    group_pk = group.pk
+    groups_data = get_group_to_permissions_and_users_mapping()
+    not_manageable_permissions = set(groups_data[group_pk]["permissions"])
+
+    # get users from groups with manage staff and look for not_manageable_permissions
+    # if any of not_manageable_permissions is found it is removed from set
+    manage_staff_users = get_users_and_look_for_permissions_in_groups_with_manage_staff(
+        group_pk, groups_data, not_manageable_permissions
+    )
+
+    # check if management of all permissions provided by other groups
+    if not not_manageable_permissions:
+        return set()
+
+    # check lack of users with manage staff in other groups
+    if not manage_staff_users:
+        return not_manageable_permissions
+
+    # look for remaining permissions from not_manageable_permissions in user with
+    # manage staff permissions groups, if any of not_manageable_permissions is found
+    # it is removed from set
+    look_for_permission_in_users_with_manage_staff(
+        group_pk, groups_data, manage_staff_users, not_manageable_permissions
+    )
+
+    # return remaining not managable permissions
+    return not_manageable_permissions
+
+
+def get_group_to_permissions_and_users_mapping():
+    """Return group mapping with data about their permissions and user.
+
+    Get all groups and return mapping in structure:
+        {
+            group1_pk: {
+                "permissions": ["perm_codename1", "perm_codename2"],
+                "users": [user_pk1, user_pk2]
+            },
+        }
+    """
+    mapping = {}
+    groups_data = (
+        Group.objects.all()
+        .annotate(
+            perm_codenames=ArrayAgg(
+                Concat(
+                    "permissions__content_type__app_label",
+                    Value("."),
+                    "permissions__codename",
+                ),
+                filter=Q(permissions__isnull=False),
+            ),
+            users=ArrayAgg("user", filter=Q(user__is_active=True)),
+        )
+        .values("pk", "perm_codenames", "users")
+    )
+
+    for data in groups_data:
+        mapping[data["pk"]] = {
+            "permissions": data["perm_codenames"],
```

Comment:
In few other places where you use this entire data structure, you need to make `set` from either `permissions` or `users`. Can we do the `set` here for both lists so we don't have to do it later when we use this structure?